ipaserver/install/ipa_cert_fix.py

```diff
@@ -228,6 +236,69 @@ def print_cert_info(context, desc, cert):
     print()
 
 
+def get_csr_from_certmonger(nickname):
+    """
+    Get the csr for the provided nickname by asking certmonger.
+
+    Returns the csr in ASCII format without the header/footer in a single line
+    or None if not found.
+    """
+    criteria = {
+        'cert-database': paths.PKI_TOMCAT_ALIAS_DIR,
+        'cert-nickname': nickname,
+    }
+
+    id = certmonger.get_request_id(criteria)
+    if id:
+        csr = certmonger.get_request_value(id, "csr")
+        if csr:
+            csr_obj = crypto_x509.load_pem_x509_csr(
+                csr.encode('ascii'), default_backend())
+            val = base64.b64encode(csr_obj.public_bytes(x509.Encoding.DER))
+            return val.decode('ascii')
```

Comment:
At first glance this seems like overkill, to load the cert, convert it between formats, etc, but it's a belt-and-suspenders approach. In the end, if there is a CSR in certmonger, we'll know that it is valid.

I think a big try/except would be good around this block though in case the CSR can't be loaded or one of the encode/decode fails.